pkg/ring/kv/etcd/etcd.go

```diff
@@ -34,10 +42,39 @@ func (cfg *Config) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.Var((*flagext.StringSlice)(&cfg.Endpoints), prefix+"etcd.endpoints", "The etcd endpoints to connect to.")
 	f.DurationVar(&cfg.DialTimeout, prefix+"etcd.dial-timeout", 10*time.Second, "The dial timeout for the etcd connection.")
 	f.IntVar(&cfg.MaxRetries, prefix+"etcd.max-retries", 10, "The maximum number of retries to do for failed ops.")
+	f.BoolVar(&cfg.EnableTLS, prefix+"etcd.tls-enabled", false, "Enable TLS")
+	f.StringVar(&cfg.CertFile, prefix+"etcd.tls-cert-file", "", "The TLS certificate file path")
+	f.StringVar(&cfg.KeyFile, prefix+"etcd.tls-key-file", "", "The TLS private key file path")
+	f.StringVar(&cfg.TrustedCAFile, prefix+"etcd.tls-client-ca-file", "", "The trusted CA file path")
```

Comment:
Sorry for being annoying but I would like to keep consistency with existing TLS config (eg. see https://cortexmetrics.io/docs/configuration/configuration-file/#ruler_config).

```suggestion
	f.StringVar(&cfg.CertFile, prefix+"etcd.tls-cert-path", "", "The TLS certificate file path")
	f.StringVar(&cfg.KeyFile, prefix+"etcd.tls-key-path", "", "The TLS private key file path")
	f.StringVar(&cfg.TrustedCAFile, prefix+"etcd.tls-ca-path", "", "The trusted CA file path")
```